blog/2025/03/this-month-in-julia-world.md

```diff
@@ -0,0 +1,63 @@
+* JETLS.jl: New [language server for Julia](https://julialang.slack.com/archives/C6FGJ8REC/p1739952975346129) with advanced static analysis in a nascent stage of development and potentially Google Summer of Code positions open for it.
+* [“C codegen considered unnecessary”](https://arxiv.org/abs/2502.01128) is a neat example by folks at JuliaHub of the new compile-to-binary capabilities introduced in Julia
+* The `@threads` macro can cause normal looking Julia code to have somewhat confusing behavior in an edge case. Namely, redefining variables inside of the newly defined closure and the way that boxing of variables works in Julia break the mental model of “a variable in Julia is just a label to a piece of actual data and assigning a variable does not modify the data, rather it just changes the what is being labeled”. A pretty esoteric source of bugs, with a [possible fix/warning proposed in a recent PR](https://github.com/JuliaLang/julia/pull/57185).
+* Pkg now prints if the [precompiled and loaded versions of a library are mismatched](https://julialang.slack.com/archives/C03D578FCDD/p1735656334706059) (frequent source of repeated precompilation when updating packages in a long-lived session). [PR1](https://github.com/JuliaLang/julia/pull/56901) and [PR2](https://github.com/JuliaLang/julia/pull/56926)
+* Compiler.jl, the julia compiler now factored out as a stdlib that you can experiment with, is getting its [best practices document](https://github.com/JuliaLang/julia/pull/57520).
+* More [caching capabilities for Julia code](https://github.com/JuliaLang/julia/pull/57193) running through customized compilers / AbstractInterpreters, remind me of the much bigger project of setting up a reliable [API for integrating external compilers](https://github.com/JuliaLang/julia/pull/52964).
+* Deep [rework and cleanup for `IOBuffer`](https://github.com/JuliaLang/julia/pull/57570) is coming up.
+* In case you did not know, besides the subtyping comparator `<:` we have a supertyping comparator `:>` that can be used in type parameterizations. It now also has a little bit more [syntactic sugar support](https://github.com/JuliaLang/julia/pull/57554).
+* Suggested best practices for [documenting return value types](https://github.com/JuliaLang/julia/pull/57583).
+* BinaryBuilder.jl will soon be able to run on newer versions of julia instead of [being stuck to 1.7](https://github.com/JuliaPackaging/JLLPrefixes.jl/issues/6) thanks to [stability guarantees to some internals of Pkg](https://github.com/JuliaLang/Pkg.jl/pull/4156) enabled by [recent work](https://github.com/JuliaLang/Pkg.jl/pull/4151).
+* Internal support for LLVM 19 and 20 in the julia compiler is [nearly done](https://github.com/JuliaLang/julia/pull/57352).
```

Comment:
LLVM 20 support is still far away at the moment. Might be more interesting to link https://github.com/JuliaLang/julia/pull/56130 saying that support for llvm 19 is now compilate and that's slated to be in Julia v1.13 (unless we migrate to llvm 20 before the feature freeze, but that's totally unpredictable).